### DISCUSSION TITLE
IDH - Interface configuration?

### DISCUSSION THREAD

**Discussion by MDCAllan** [Security-Onion-Solutions/securityonion#11903] [Unsupported Versions] [answered]
Hello!

We are deploying a new SO 2.4 grid. So far things are going pretty well. Our deployment is a total of 15 nodes including an IDH node. We just need a little help sorting out some of the finer details where I assume 2.4 is still working out some kinks and/or needs more documentation. Keep up the great work!

When I read the documentation for the IDH, I'm operating on the impression that the additional non-management interfaces are supposed to "auto configure" themselves with open ports, etc. We have DCHP reservations for all 5 of the additional interfaces on the IDH node and those interfaces are attached to the networks we want to deploy honeypots to, but none of the interfaces are coming online, pulling a lease, or opening the honey ports. The documentation does not explain how to assign IP's or configure DHCP for the additional interfaces. I'm not sure if this should be done according to rocky linux standards, or if there's a salt/yaml/config thing that needs to be done using an SO command or special file that needs editing. I don't want to break something by poking around too much as this is a purpose built system.

Any help greatly appreciated, also, great job on 2.4! We're super excited to deploy agents and start hunting! Already have 4 forward nodes successfully monitoring 4 gateways in the environment with 2.4, and in this deployment we scaled out to 6 Search nodes to allow better load balancing across our server cluster. Fantastic!

Thanks!

**Comment by defensivedepth** [comment]
IDH services will run on all IP-configured NICs. There is no auto-configure in place for NICs other than the NIC that is selected as the Management NIC during setup. You should be able to set up the others manually without issue.

**Reply by MDCAllan** [reply, marked as answer]
Thanks Defensivedepth!

The "nmtui" tool included in Rocky 9 makes it easy to activate and configure interfaces (and enable autoconnect). Might be worth mentioning this in the SO Docs. 

Indeed the interfaces are honeypotting now! Thanks!

**Reply by dougburks** [reply]
> The "nmtui" tool included in Rocky 9 makes it easy to activate and configure interfaces (and enable autoconnect). Might be worth mentioning this in the SO Docs.

I've updated the docs:
https://docs.securityonion.net/en/2.4/idh.html#activating-additional-network-interfaces

**Comment by MDCAllan** [comment]
Nice!

Getting closer... I enabled the IDH playbooks and we started hammering the various honeypot services and not seeing any alerts...

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/77131271/c238ccf8-0f9b-45e6-b9da-6cd7bd2801d7)

I suspect it has something to do with this... I just don't know what "word" it wants... "false" seems to be the default, but true is not an option. 

FYI I love the GUI configuration tools on the new SO web interface. This is so much better for guys like me who can't handle yamls very well ;)

**Reply by defensivedepth** [reply]
You should not need to do any configuration for Playbook - within 15 minutes of the first IDH node joining the Grid, Playbook will automatically import the baseline IDH Plays. If you navigate over to Playbook, you should see them - you can search for "IDH"

**Reply by MDCAllan** [reply]
The plays were imported but not active. I activated them last week but still no alerts when we attempt sign-ins / brute force on the honeypot. 

We have a handful of Elastic Agents deployed, which, usually would generate some routine/low alerts in normal use and we're seeing nothing there either. I can hunt and find event logs associated with the new agents, but they are not showing up as observers in dashboard the way they did in SO 2.3 with winlogbeat deployed. 

I think there's something broken with our deployment. 

I wonder... When I deployed this grid, starting with the manager, I selected "FQDN" instead of "hostname" during the initial sequence of questions. I did this on purpose because this new deployment was expected to have nodes spread out on multiple networks. I wonder if this is causing other problems. I've always used hostname only in the past.

**Reply by defensivedepth** [reply]
They should have been automatically enabled. Can you confirm there are corresponding elastalert rules in `/opt/so/rules/playbook/` for them? Also, can you confirm that there are IDH logs that you can query via the Hunt interface?

There were alot of low level / info 'alerts' generated via Wazuh in 2.3. With Wazuh no longer included in 2.4, you will not get these low level / info alerts. Is this possibly what you are thinking of?

**Reply by MDCAllan** [reply]
They were not automatically enabled. 

We're no longer running Wazuh, we went to a sysmon+winlogbeats approach for our 2.3 deployment about a year ago since we knew Wazuh was going away and were having problems with OSQUERY (the OSQUERY service was causing teams to crash). We still see routine stuff like "Logon Success" and " Audit Failure event" and "Logoff" and "The VSS service is shutting down due to idle timeout" etc...show up as alerts... And normal activity in our domain that triggers certain playbooks. 

We're getting only suricata alerts right now on the 2.4 deployment, no alerts from any other modules, seems strange. 

There are some yaml files in /opt/so/rules/elastalert/playbook/ ... I would estimate that the number of yaml files here matches the number of plays that are active right now. 

If I enter "IDH" in the hunt query interface I see a lot of system.syslog logs from the IDH node...

lots of: 
[  "failed in Fleet agent final_pipeline: field [created] not present as part of path [event.created]" ]

**Reply by defensivedepth** [reply]
A couple things:

- Do you have any Plays enabled other than the IDH ones?
- Tail the elastalert logs and see if there are any errors: `sudo tail -f /opt/so/log/elastalert/elastalert.log`

**Reply by MDCAllan** [reply]
Yea I selected a handful of plays to throw in the mix for starters. I searched the plays and activated all plays with the keyword "install" - Combined with the IDH plays that's about 100 active plays. We currently have about 1500 active plays on our 2.3 deployment and that seems to be working. I think you're on the right track though... Thanks so much for the continued assistance on this! 

Found some errors:

```
WARNING        elasticsearch POST https://soman:9200/.ds-logs-*/_eql/search?ignore_unavailable=true [status:400 request:0.011s]  
ERROR           elastalert Error running query: RequestError(400, 'verification_exception', 'Found 1 problem\nline 1:12: first argument of ["Install-TransportAgent" and winlog.channel : "MSExchange Management"] must be [boolean], found value ["Install-TransportAgent"] type [keyword]')

WARNING        elasticsearch POST https://soman:9200/.ds-logs-*/_eql/search?ignore_unavailable=true [status:400 request:0.011s]
ERROR           elastalert Error running query: RequestError(400, 'verification_exception', 'Found 1 problem\nline 1:82: Unknown column [winlog_channel]')

WARNING        elasticsearch POST https://soman:9200/.ds-logs-*/_eql/search?ignore_unavailable=true [status:400 request:0.012s]
ERROR           elastalert Error running query: RequestError(400, 'verification_exception', 'Found 1 problem\nline 1:13: first argument of [:] must be [string], found value [process.parent.pid] type [long]; consider using [==] instead')

WARNING        elasticsearch POST https://soman:9200/.ds-logs-*/_eql/search?ignore_unavailable=true [status:400 request:0.013s]
ERROR           elastalert Error running query: RequestError(400, 'planning_exception', 'Found 20 problems\nline 1:36: Unresolved expression\nline 1:92: Unresolved expression\nline 1:149: Unresolved expression\nline 1:198: Unresolved expression\nline 1:260: Unresolved expression\nline 1:36: Unresolved expression\nline 1:92: Unresolved expression\nline 1:149: Unresolved expression\nline 1:198: Unresolved expression\nline 1:260: Unresolved expression\nline 1:36: Unresolved expression\nline 1:92: Unresolved expression\nline 1:149: Unresolved expression\nline 1:198: Unresolved expression\nline 1:260: Unresolved expression\nline 1:36: Unresolved expression\nline 1:92: Unresolved expression\nline 1:149: Unresolved expression\nline 1:198: Unresolved expression\nline 1:260: Unresolved expression')
```

I deactivated plays with "msexchange" in their title and that first error went away. Not sure what to do with the others.

**Reply by defensivedepth** [reply]
With the move from Lucene to EQL, there are still some field mapping issues. That's what I am seeing in those logs. It is on our list to dig into them deeper soon.

I would suggest you focus on a few key Plays, enable them, and confirm that they are generating alerts as expected.

To be clear, these shouldnt impact the IDH alerts - I have confirmed those should be working as expected.

**Reply by MDCAllan** [reply]
I rebuilt the new grid from scratch (wiped all the virtual drives and reinstalled all nodes). 

Alerts from IDH nodes are working now. I have no idea what was corrupted with the previous grid but it probably wasn't worth doing battle with any longer. Thanks so much for all of your help. All a great learning experience for me. 

We deployed our new grid with 4 separate honeypots, each with "tasty" hostnames and each with multiple interfaces on separate networks. Better this way anyway. 

Luckily we deploy on proxmox so rebuilding the grid is just a lot of busywork pushing buttons for a day.

**Reply by defensivedepth** [reply]
Good to hear! Happy hunting!
